packages/amplify-migration-tests/src/__tests__/migration_tests_v10/auth-add-all.migration.test.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    addAuthWithDefault,
+    addAuthWithMaxOptions,
+    amplifyPull,
+    amplifyPushAuth,
+    amplifyPushWithoutCodegen,
+    createNewProjectDir,
+    deleteProject,
+    deleteProjectDir,
+    getAppId,
+    removeAuthWithDefault,
+  } from '@aws-amplify/amplify-e2e-core';
+import { versionCheck, allowedVersionsToMigrateFrom } from '../../migration-helpers';
+import { initJSProjectWithProfileV10 } from '../../migration-helpers-v10/init';
+import { assertNoParameterChangesBetweenProjects, collectCloudformationDiffBetweenProjects, ExcludeFromCFNDiff } from '../../migration-helpers/utils';
+
+/**
+ * Due to a known limitation of APIGateway, we do not want the AWS::ApiGateway::Deployment resource
+ * from being compared in the CFN diff because it is regenerated whenever an amplify app is pulled down.
+ * This would produce a false positive when checking for differences in the CFN templates of project1 & project2.
+ * https://github.com/aws/aws-cdk/issues/8646#issuecomment-647561856
+ */
```

Comment:
Below is an example of how the resource changes even though the projects are identical
![Screen Shot 2022-12-21 at 8 06 07 PM](https://user-images.githubusercontent.com/110861985/209033567-af750afb-5255-4216-8c70-a746053e006d.png)

Reply:
The code (below) deletes this property from the resource's CFN template before we do the comparison.